ksqldb-engine/src/main/java/io/confluent/ksql/engine/InsertValuesExecutor.java

```diff
@@ -292,29 +299,39 @@ private void throwIfDisabled(final KsqlConfig config) {
     try {
       return valueSerde.serializer().serialize(topicName, row);
     } catch (final Exception e) {
-      final Format valueFormat = FormatFactory
-          .fromName(dataSource.getKsqlTopic().getValueFormat().getFormat());
-      if (valueFormat.supportsFeature(SerdeFeature.SCHEMA_INFERENCE)) {
-        final Throwable rootCause = ExceptionUtils.getRootCause(e);
-        if (rootCause instanceof RestClientException) {
-          switch (((RestClientException) rootCause).getStatus()) {
-            case HttpStatus.SC_UNAUTHORIZED:
-            case HttpStatus.SC_FORBIDDEN:
-              throw new KsqlException(String.format(
-                  "Not authorized to write Schema Registry subject: [%s]",
-                  topicName + KsqlConstants.SCHEMA_REGISTRY_VALUE_SUFFIX
-              ));
-            default:
-              break;
-          }
-        }
-      }
-
+      maybeThrowSchemaRegistryAuthError(
+          FormatFactory.fromName(dataSource.getKsqlTopic().getValueFormat().getFormat()),
+          topicName,
+          false,
+          e);
       LOG.error("Could not serialize row.", e);
```

Comment:
nit (not your code): can we update the error message here to say `Could not serialize value`? To me "row" could refer to either the key or the value. Seems good to be more specific when we can.

Reply:
Since I need to update the other PR and this one is green, i'm going to just go ahead and merge and do it in https://github.com/confluentinc/ksql/pull/6452